core/src/main/java/org/apache/iceberg/deletes/Deletes.java

```diff
@@ -191,7 +197,12 @@ private PositionStreamDeleteFilter(CloseableIterable<T> rows, Function<T, Long>
       return iter;
     }
 
-    private class PositionFilterIterator extends FilterIterator<T> {
+    protected FilterIterator<T> getPositionIterator(CloseableIterator<T> items,
+                                                    CloseableIterator<Long> newDeletePositions) {
+      return new PositionFilterIterator(items, newDeletePositions);
+    }
+
+    protected class PositionFilterIterator extends FilterIterator<T> {
```

Comment:
Does this need to be protected? It looks like it could be private and still used. Or, it could be protected and the `getPositionIterator` method could be removed. I think I prefer the second option because it would require no change to the `iterator` method in this class.